src/prefect/engine/executors/dask.py

```diff
@@ -271,7 +340,14 @@ def submit(
             raise ValueError("This executor has not been started.")
 
         kwargs.update(self._prep_dask_kwargs(extra_context))
-        return self.client.submit(fn, *args, **kwargs)
+        if self._should_run_var is None:
+            fut = self.client.submit(fn, *args, **kwargs)
+        else:
+            fut = self.client.submit(
+                _maybe_run, self._should_run_var, fn, *args, **kwargs
+            )
+            self._futures.add(fut)
```

Comment:
When I first saw this I was worried about a situation where this was `None` but after a re-read it appears that `_should_run_var` and `_futures` are coupled attributes - any way to simplify that?  

I'm worried about the case where lines 255-256 are refactored / initialized somewhere else without realizing the dependency.

Reply:
I don't see a cleaner way to refactor this - they're coupled attributes, but that's what tests are for? Things will only pass with the current implementation working the way it does. I can add a comment somewhere I suppose.

Reply:
Coupled hidden attributes make me nervous but I'm definitely not interested in holding the PR up for it - the comment helps 👍